### PR TITLE
136 activities to display the notifications

### PR DESCRIPTION
**Pull request by TheGuyWithoutH**
Add an activity to show notifications received. To be quicker in the tasks, another PR will later arrive for actions when clicking on the notif, as well as a cleaning of old notifications.

image

### CODE REVIEW

File: app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+public class NotificationsRecycleViewAdapter extends RecyclerView.Adapter<NotificationsRecycleViewAdapter.NotificationViewHolder> {
+
+    private List<PushNotification> notificationTexts;
+    private final String UID;
+
+    public NotificationsRecycleViewAdapter(NotificationsViewModel notificationsViewModel) {
+        notificationsViewModel.getNotificationTexts().observeForever(notificationTexts -> {
+            this.notificationTexts = notificationTexts;
+            System.out.println("Notifications: " + notificationTexts.stream().map(PushNotification::getText).collect(Collectors.toList()));
```
Comment (lucamouchel) [inline app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java:34]: make sure to remove this before merging ;)

File: app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    @Override
+    public int getItemCount() {
+        return notificationTexts.size();
+    }
+
+    private void setIconNotification(ImageView icon, String channel) {
+        switch (channel) {
+            case ScanNotification.CHANNEL_ID:
+                icon.setImageResource(R.drawable.scan_icon_unsel);
+                break;
```
Comment (lucamouchel) [inline app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java:67]: when is the scan notification used? is it for offline mode? or for sth else? bc if it's for every scan isn't the notification a bit overkill?

File: app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_notification, parent, false);
+        return new NotificationViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull NotificationsRecycleViewAdapter.NotificationViewHolder holder, int position) {
+        holder.getNotificationText().setText(notificationTexts.get(position).getText());
+        setIconNotification(holder.getNotificationIcon(), notificationTexts.get(position).getChannelId());
+        holder.getDeleteButton().setOnClickListener(view -> {
+            Database.deleteNotification(UID, notificationTexts.get(position));
```
Comment (lucamouchel) [inline app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java:52]: maybe just inline UID with profile.getActive.getUid insted of having a UID variable

Comment (lucamouchel) [reply app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java:52]: you only use UID once in this class

File: app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsViewModel.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import ch.epfl.culturequest.notifications.PushNotification;
+import ch.epfl.culturequest.social.Profile;
+
+public class NotificationsViewModel extends ViewModel {
+    private final MutableLiveData<List<PushNotification>> notificationTexts;
+
+    public NotificationsViewModel() {
+        notificationTexts = new MutableLiveData<>();
+        String currentUserUid = Profile.getActiveProfile().getUid();
+        Database.getNotifications(currentUserUid).whenComplete((notifications, throwable) -> {
```
Comment (lucamouchel) [inline app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsViewModel.java:19]: same here, maybe just inline currentUserUid

**Review by lucamouchel — COMMENTED** [review]
Lgtm, just some minor stuff

File: app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java (hunk 2 above)
Comment (TheGuyWithoutH) [reply app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java:67]: I cannot tell, I only used the types already created by ahmed and hugo.

**Review by lucamouchel — APPROVED** [review]

**Review by lucamouchel — APPROVED** [review]

**Pull request merged**